src/CarbonAware.Aggregators/src/CarbonAware/ICarbonAwareAggregator.cs

```diff
@@ -21,10 +21,18 @@ public interface ICarbonAwareAggregator : IAggregator
 
 
     /// <summary>
-    /// Get forecasted emissions data.
+    /// Get current forecasted emissions data.
     /// </summary>
     /// <param name="props">IDictionary with properties required by concrete classes.</param>
     /// <see cref="CarbonAwareConstants"/>
     /// <returns>List of current emissions forecasts by location.</returns>
     Task<IEnumerable<EmissionsForecast>> GetCurrentForecastDataAsync(IDictionary props);
+
+    /// <summary>
+    /// Get forecasted emissions data.
+    /// </summary>
+    /// <param name="props">IDictionary with properties required by concrete classes.</param>
+    /// <see cref="CarbonAwareConstants"/>
+    /// <returns>Single emissions forecast for a given location at the requestedAt time given start and end periods.</returns>
+    Task<EmissionsForecast> GetForecastDataAsync(IDictionary props);
```

Comment:
New function for getting a forecast given a generatedAt time and optionally forecastStart/forecastEnd fields for filtering